compiler/desugared/from_surface.ml

```diff
@@ -269,53 +266,29 @@ let rec translate_expr
         (SubScopeVar
            (subscope_real_uid, (subscope_uid, pos), (subscope_var_uid, pos)))
         emark
-    | _ -> (
+    | _ ->
       (* In this case e.x is the struct field x access of expression e *)
       let e = translate_expr scope inside_definition_of ctxt e in
-      let x_possible_structs =
-        try Name_resolution.IdentMap.find (Marked.unmark x) ctxt.field_idmap
-        with Not_found ->
-          Errors.raise_spanned_error (Marked.get_mark x)
-            "Unknown subscope or struct field name"
+      let str =
+        Option.map
+          (fun c ->
+            try Name_resolution.get_struct ctxt c
+            with Not_found ->
+              Errors.raise_spanned_error (Marked.get_mark c)
+                "Struct %s has not been defined before" (Marked.unmark c))
```

Comment:
The error message suggest you must declare (not define) structs before using them. But actually this is not the case since the closing of https://github.com/CatalaLang/catala/issues/47.

Reply:
Ah indeed, I should have taken the opportunity to fix it!